ampersand-model.js

```diff
@@ -60,10 +89,18 @@ var Model = State.extend({
         wrapError(this, options);
 
         method = this.isNew() ? 'create' : (options.patch ? 'patch' : 'update');
-        if (method === 'patch') options.attrs = attrs;
+
+        if (method === 'patch') {
+            options.attrs = transformForPatch(this.serialize(), attrs);
+        }
         // if we're waiting we haven't actually set our attributes yet so
         // we need to do make sure we send right data
-        if (options.wait && method !== 'patch') options.attrs = assign(model.serialize(), attrs);
+        if (options.wait && method !== 'patch') {
+            var clonedModel = new this.constructor(this.getAttributes({props: true}));
+            clonedModel.set(attrs);
```

Comment:
lol, correct. 

The issue with `this.serialize()` was because i was ignoring the top-level object up in `transformForPatch`. I pulled the `intersection` stuff out of the `transform` so that it always runs against the passed-in object first.